vectorstores/pgvector/engine.go

```diff
@@ -0,0 +1,141 @@
+package pgvector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/oauth2/v2"
+	"google.golang.org/api/option"
+)
+
+const Port = 5432 // default PostgreSQL port
+
+type ServiceAccountRetriever interface {
+	GetServiceAccountEmail(ctx context.Context) (string, error)
+}
+type DefaultServiceAccountRetriever struct{}
+
+func (d *DefaultServiceAccountRetriever) GetServiceAccountEmail(ctx context.Context) (string, error) {
+	return getServiceAccountEmail(ctx)
+}
+
+type Column struct {
+	Name     string
+	DataType string
+	Nullable bool
+}
+
+func NewColumn(name, dataType string, nullable bool) (*Column, error) {
+	if name == "" {
+		return nil, errors.New("name should be provided")
+	}
+	if dataType == "" {
+		return nil, errors.New("dataType should be provided")
+	}
+	return &Column{Name: name, DataType: dataType, Nullable: nullable}, nil
+}
+
+type PostgresEngine struct {
+	Host                    string
+	Port                    int
+	Database                string
+	User                    string
+	Password                string
+	IAMAccountEmail         string
+	ServiceAccountRetriever ServiceAccountRetriever
+}
```

Comment:
This doesn't look like an engine -- it looks like a configuration that could be used to create an engine. I would expect the engine to hold a reference to a pool. 

Separately, similar to my other comments we should probably choose one between initializer and exported fields.

Reply:
You are right! I changed the name to a more appropiate one. 
The reference to a pool is something like the one on pgvector.go https://github.com/tmc/langchaingo/blob/main/vectorstores/pgvector/pgvector.go#L57 with the pool corresponding methods?
And can you elaborate on the initializer and exported fields please?